pkg/paillier/secret.go

```diff
@@ -154,44 +153,47 @@ func (sk SecretKey) Validate() error {
 
 	// check bit lengths
 	const bitsWant = params.BitsBlumPrime
-	if bits := sk.p.BitLen(); bits != bitsWant {
+	// Technically, this leaks the number of bits, but this is fine, since returning
+	// an error asserts this number statically, anyways.
+	if bits := sk.p.TrueLen(); bits != bitsWant {
 		return fmt.Errorf("paillier.SecretKey: prime p have: %d, need %d: %w", bits, bitsWant, ErrPrimeBadLength)
 	}
-	if bits := sk.q.BitLen(); bits != bitsWant {
+	if bits := sk.q.TrueLen(); bits != bitsWant {
 		return fmt.Errorf("paillier.SecretKey: prime q have: %d, need %d: %w", bits, bitsWant, ErrPrimeBadLength)
 	}
 
+	// Hopefully this doesn't leak too much information about p or q
 	// check prime
-	if !sk.p.ProbablyPrime(20) {
+	if !sk.p.Big().ProbablyPrime(20) {
 		return fmt.Errorf("paillier.SecretKey: prime p: %w", ErrNotPrime)
 	}
 	// check prime
-	if !sk.q.ProbablyPrime(20) {
```

Comment:
Now that we have safe primes, we could also verify that p-1/2 is also prime?

But maybe we should decouple the prime checking to another method since we wouldn't want to check this every time a signature is generated. Only really needs to be done when the data is being unmarshalled

Reply:
Might be a good idea to have a "quick" and "thoroughly" validate, I guess